### PR TITLE
change title checks to use regex matching instead of substring matching

### PR DESCRIPTION
**Pull request by ewlu**

### CODE REVIEW

File: scripts/get_baseline_hash.py [hunk 1]
```diff
@@ -21,7 +22,8 @@ def filter_results(issue):
         if label['name'] in filter_labels:
             labels_check = False
             break
-    title_check = "Testsuite Status" in issue["title"]
+    regex_check = re.search("^Testsuite Status.*$", issue["title"]) # if fails, regex_check == None
```
Comment (patrick-rivos) [inline scripts/get_baseline_hash.py:25]: I think we can drop the `.*$` and get the same behavior

File: scripts/get_baseline_hash.py [hunk 2]
```diff
@@ -21,7 +22,8 @@ def filter_results(issue):
...
+    title_check = True if regex_check is not None else False
```
Comment (patrick-rivos) [inline scripts/get_baseline_hash.py:26]:
```suggestion
    title_check = regex_check is not None
```

File: scripts/get_baseline_hash.py (hunk 1 above)
Comment (patrick-rivos) [inline scripts/get_baseline_hash.py:25]: I think we can get away with this to make the comparison a single statement:
```
    title_check = re.search("^Testsuite Status", issue["title"]) is not None # if the regex search fails, re.search returns None
```

**Review by patrick-rivos — CHANGES_REQUESTED** [review]

File: scripts/get_baseline_hash.py (hunk 1 above)
Comment (ewlu) [reply scripts/get_baseline_hash.py:25]: just double checked and yeah we can

Comment (ewlu) [reply scripts/get_baseline_hash.py:25]: this also works

**Review by patrick-rivos — APPROVED** [review]
LGTM

**Pull request merged**